src/Symfony/Component/DependencyInjection/Compiler/ResolveChildDefinitionsPass.php

```diff
@@ -111,6 +111,7 @@ private function doResolveDefinition(ChildDefinition $definition): Definition
         $def->setLazy($parentDef->isLazy());
         $def->setAutowired($parentDef->isAutowired());
         $def->setChanges($parentDef->getChanges());
+        $def->setSynthetic($parentDef->isSynthetic());
```

Comment:
the flag should not be borrowed from $parentDef, but from $definition instead

Reply:
Ok, updated.